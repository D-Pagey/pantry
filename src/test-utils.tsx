import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import React, { FC, ReactNode } from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { FirebaseContext } from './components/ProviderFirebase';

type AllTheProvidersProps = {
    children: ReactNode;
    customContext?: any;
};

const AllTheProviders: FC<AllTheProvidersProps> = ({ children, customContext }) => (
    <FirebaseContext.Provider
        value={{
            categories: [],
            deleteFoodItem: () => () => null,
            expiringCount: 0,
            isAuthed: false,
            isCheckingAuth: false,
            fridge: [],
            signOut: () => null,
            updateFridge: () => null,
            user: {
                email: null,
                name: null
            },
            ...customContext
        }}
    >
        <MemoryRouter>{children}</MemoryRouter>
    </FirebaseContext.Provider>
);

const customRender = (ui: any, customContext?: any, options?: any) =>
    render(ui, { wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
