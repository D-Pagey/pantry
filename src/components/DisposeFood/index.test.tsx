import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../test-utils';
import { DisposeFood } from '.';

const props = {
    handleDelete: () => {},
    handleEdit: () => {},
    name: 'carrot'
};

describe('DisposeFood component', () => {
    it('should render', () => {
        const { container } = render(<DisposeFood {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call handleDelete when eat clicked', () => {
        const handleDelete = jest.fn();
        const { getByText } = render(<DisposeFood {...props} handleDelete={handleDelete} />);

        userEvent.click(getByText("Eat all carrot's"));

        expect(handleDelete).toHaveBeenCalled();
    });

    it('should call handleEdit when edit clicked', () => {
        const handleEdit = jest.fn();
        const { getByText } = render(<DisposeFood {...props} handleEdit={handleEdit} />);

        userEvent.click(getByText('Edit servings'));

        expect(handleEdit).toHaveBeenCalled();
    });
});
