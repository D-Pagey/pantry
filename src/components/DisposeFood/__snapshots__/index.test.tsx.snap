// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisposeFood component should render 1`] = `
.c1 {
  background-color: #348841;
  border: 0;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  padding: 1rem;
  min-width: 68px;
}

.c2 {
  background-color: white;
  border: 1px solid #348841;
  color: #348841;
}

.c0 {
  background-color: white;
  border-top: 1px solid #B2B2B2;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  left: 0;
  padding: 1.5rem 0;
  position: fixed;
  right: 0;
}

<div
  class="c0"
  data-testid="disposeFood"
>
  <button
    class="c1 c2"
    type="button"
  >
    Edit servings
  </button>
  <button
    class="c1"
    type="button"
  >
    Eat all 
    carrot
    's
  </button>
</div>
`;
